src/OSPSuite.Core/Domain/Services/PKValuesCalculator.cs

```diff
@@ -268,25 +310,44 @@ private class DosingInterval
          public double Vss => CL * Mrt;
          public double Vd => CL / _lambda;
 
-         public DosingInterval(List<float> time, List<float> conc, PKCalculationOptions options)
+         public PKInterval(List<float> time, List<float> concentration, PKCalculationOptions options, double? dose = null, float? concentrationThreshold = null)
          {
             _options = options;
+            Dose = dose;
             _time = time;
-            _concentration = conc;
+            _concentration = concentration;
+            _concentrationThreshold = concentrationThreshold;
             _timeSteps = ArrayHelper.TimeStepsFrom(time);
             _polyFit = new PolyFit();
          }
 
          public void Calculate()
          {
             Cmax = _concentration.Max();
+            Cmin = _concentration.Min();
             Tmax = _time[_concentration.IndexOf(Cmax)];
+            Tmin = _time[_concentration.IndexOf(Cmin)];
             CTrough = _concentration.Last();
             Auc = calculateAuc();
             calculateAucInf();
-
             Aucm = calculateAucm();
             Mrt = calculateMrt();
+            Tthreshold = calculateTThreshold();
+         }
+
+         private float calculateTThreshold()
+         {
+            if (_concentrationThreshold == null)
+               return float.NaN;
+
+            var startIndex = _concentration.IndexOf(Cmax);
```

Comment:
I am starting a Cmax. I am not sure if this is right but otherwise, for Oral application, we might hit the threshold on the absorption phase? Or should we just start from zero?

Reply:
Think it's ok. I wouldn't start from zero indeed